network/p2p/client.go

```diff
@@ -41,6 +41,7 @@ type Client struct {
 	handlerPrefix []byte
 	router        *Router
 	sender        common.AppSender
+	nodeSampler   NodeSampler
```

Comment:
So I could potentially see some confusion where we now expose an interface that allows us to choose our sampling behavior. This is distinct from `AppGossip`, which still uses the config passed when the underlying `sender` was created.

Reply:
Comments never hurt :)

Reply:
Or we could have `AppGossip` use the `NodeSampler` -- that would seem reasonable to me

Reply:
issue: https://github.com/ava-labs/avalanchego/issues/1890

Reply:
In that case we should probably put a comment explaining when `nodeSampler` is used and when the sender's config is used